### PR TITLE
Open Data Portal Cronjob

### PR DESCRIPTION
**Pull request by 1AhmedYasser**
Related Task:

- https://github.com/buerokratt/Analytics-Module/issues/131

### CODE REVIEW

File: DSL/Ruuter/POST/odp/delete-cron-job-task.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+deleteFile:
+  call: http.post
+  args:
+    url: "[#ANALYTICS_DMAPPER]/file-manager/delete"
+    body:
+      file_path: ${incoming.body.location}
+  result: fileResult
+
+refereshCronManagerIfChanged:
```
Comment (ValterAro) [inline DSL/Ruuter/POST/odp/delete-cron-job-task.yml:9]: Small typo, should be - > "refreshCronManagerIfChanged"

File: DSL/Ruuter/POST/odp/delete-scheduled-report.yml [hunk 2]
```diff
@@ -6,22 +6,28 @@ getJobDetails:
...
 deleteScheduledReport:
   call: http.post
   args:
     url: "[#ANALYTICS_RESQL]/delete-scheduled-report"
     body:
-      id: ${jobDetails.response.body[0].id}
+      id: ${incoming.body.datasetId}
   result: delete
 
+refereshCronManagerIfChanged:
```
Comment (ValterAro) [inline DSL/Ruuter/POST/odp/delete-scheduled-report.yml:25]: Same typo, should be - > "refreshCronManagerIfChanged"

File: DSL/Ruuter/POST/odp/update-scheduled-report.yml [hunk 3]
```diff
@@ -68,10 +58,33 @@ saveJobInformation:
...
+delete-schedule:
+  call: http.post
+  args:
+    url: "[#ANALYTICS_RUUTER]/odp/delete-scheduled-report"
+    body:
+      datasetId: ${incoming.body.datasetId}
+  result: delete_res
+  next: refereshCronManagerIfChanged
+
+refereshCronManagerIfChanged:
```
Comment (ValterAro) [inline DSL/Ruuter/POST/odp/update-scheduled-report.yml:81]: Same typo, should be - > "refreshCronManagerIfChanged"

File: DSL/Ruuter/POST/saveJsonToYml.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      content: ${incoming.body.yaml}
+  result: fileResult
+
+check-if-cron-is-changed:
+  switch:
+    - condition: ${incoming.body.location.includes('CronManager')}
+      next: refereshCronManagerIfChanged
+  next: return_result  
+
+refereshCronManagerIfChanged:
```
Comment (ValterAro) [inline DSL/Ruuter/POST/saveJsonToYml.yml:16]: Same typo, should be - > "refreshCronManagerIfChanged"

File: DSL/Ruuter/POST/odp/update-scheduled-report.yml [hunk 5]
```diff
@@ -68,10 +58,33 @@ saveJobInformation:
...
       dataset_id: ${incoming.body.datasetId}
+      start_date: ${incoming.body.start}
+      end_date: ${incoming.body.end}
   result: scheduledReport
 
+check-for-interval:
+  switch:
+    - condition: ${incoming.body.updateIntervalUnit === 'never'}
+      next: delete-schedule
+  next: refereshCronManagerIfChanged
```
Comment (ValterAro) [inline DSL/Ruuter/POST/odp/update-scheduled-report.yml:70]: Same typo

File: DSL/Ruuter/POST/saveJsonToYml.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    url: "[#ANALYTICS_DMAPPER]/file-manager/create"
+    body:
+      file_path: ${incoming.body.location}
+      content: ${incoming.body.yaml}
+  result: fileResult
+
+check-if-cron-is-changed:
+  switch:
+    - condition: ${incoming.body.location.includes('CronManager')}
+      next: refereshCronManagerIfChanged
```
Comment (ValterAro) [inline DSL/Ruuter/POST/saveJsonToYml.yml:13]: Same typo

**Review by baha-a — APPROVED** [review]

**Pull request merged**